adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralRewardedAdLoader.m

```diff
@@ -0,0 +1,140 @@
+}
+
+#pragma mark - MTGRewardAdLoadDelegate
+- (void)onVideoAdLoadSuccess:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
+    if (_adLoadCompletionHandler) {
+        _adEventDelegate = _adLoadCompletionHandler(self, nil);
+    }
+}
+
+- (void)onVideoAdLoadFailed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId error:(nonnull NSError *)error {
+    if (_adLoadCompletionHandler) {
+        _adLoadCompletionHandler(nil, error);
+    }
+}
+
+#pragma mark - MTGRewardAdShowDelegate
+- (void)onVideoAdShowSuccess:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
+    [_adEventDelegate willPresentFullScreenView];
+    [_adEventDelegate reportImpression];
+    [_adEventDelegate didStartVideo];
+}
+
+- (void)onVideoAdShowFailed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId withError:(nonnull NSError *)error {
+    [_adEventDelegate didFailToPresentWithError:error];
+}
+
+- (void)onVideoPlayCompleted:(nullable NSString *)placementId unitId:(nullable NSString *)unitId{
+    [_adEventDelegate didEndVideo];
+}
+
+- (void)onVideoAdClicked:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
+    [_adEventDelegate reportClick];
+}
+
+- (void)onVideoAdDismissed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId withConverted:(BOOL)converted withRewardInfo:(nullable MTGRewardAdInfo *)rewardInfo {
+    [_adEventDelegate willDismissFullScreenView];
+    if (converted) {
+        [_adEventDelegate didRewardUser];
+    }
+}
+
+- (void)onVideoAdDidClosed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
+    [_adEventDelegate didDismissFullScreenView];
+}
+
+#pragma mark - GADMediationRewardedAd
+- (void)presentFromViewController:(nonnull UIViewController *)viewController {
+    NSString *adUnitId = _adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = _adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    if (!adUnitId.length || !placementId.length) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID or Placement ID cannot be nil.");
+        [_adEventDelegate didFailToPresentWithError:error];
+        return;
+    }
+    GADMAdapterMintegralExtras *extras = _adConfiguration.extras;
+    _rewardedAd = MTGBidRewardAdManager.sharedInstance;
+    _rewardedAd.playVideoMute = extras.muteVideoAudio;
+    if ([_rewardedAd isVideoReadyToPlayWithPlacementId:placementId unitId:adUnitId]) {
+        [_rewardedAd showVideoWithPlacementId:placementId unitId:adUnitId withRewardId:nil userId:nil delegate:self viewController:viewController];
+    }else{
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorAdFailedToShow, @"Failed to display rewarded video ad from Mintegral.");
```

Comment:
Nit, can you update the error message to: `Mintegral SDK failed to present a rewarded video ad.`?

Reply:
OK, I've made changes and committed.